draft-sharabayko-mops-srt.md

```diff
@@ -2105,380 +2130,210 @@ with a key establishment method.
-which counts the blocks in a packet, (2) a packet index - based on the packet sequence number
-in the UDT header - in the next 32 bits, and (3) eighty bits that are zeroes.
-The upper 112 bits of this sequence are XORed with an initialization vector (IV)
-to produce a unique counter for each crypto block. Note that the block counter is unique for 
-the duration of the session. The same counter cannot be used twice.
-
-This key used for encryption is called the "Stream Encrypting Key" (SEK), which is used for
-2^25 packets.
+a 128-bit sequence consisting of
+
+- a block counter in the least significant 16 bits, which counts the blocks in a packet,
+
+- a packet index - based on the packet sequence number in the SRT header - in the next 32 bits,
+
+- eighty zeroed bits.
+
+The upper 112 bits of this sequence are XORed with an Initialization Vector (IV)
+to produce a unique counter for each crypto block. The IV is derived from the Salt
+provided in the Keying Material ({{sec-ctrlpkt-km}}):
+
+~~~~
+IV = MSB(112, Salt): Most significant 112 bits of the salt.
+~~~~
+
+### Stream Encrypting Key (SEK)
+
+The key used for AES-CTR encryption is called the "Stream Encrypting Key" (SEK). It is used for
+up to 2^25 packets with further rekeying.
 The short-lived SEK is generated by the sender using a pseudo-random number generator (PRNG),
-and transmitted within the stream (KM Tx Period), wrapped with another longer-term key,
+and transmitted within the stream, wrapped with another longer-term key,
 the Key Encrypting Key (KEK), using a known AES key wrap protocol.
 
-<!-- TODO: [[Figure]] -->
-
 For connection-oriented transport such as SRT, there is no need to periodically transmit
 the short-lived key since no party can join the stream at any time. The keying material
 is transmitted within the connection handshake packets, and for a short period
 when rekeying occurs.
 
-The KEK is derived from a secret shared (e.g. passphrase) between the sender and the receiver.
-The shared secret provides access to the stream key, which provides access to the protected
-media stream. The distribution and management of the secret is more flexible than the stream
-encrypting key. The KEK has to be at least as long as the SEK.
+### Key Encrypting Key (KEK)
+
+The KEK is derived from a secret (passphrase) shared between the sender and the receiver.
+The Key Encrypting Key provides access to the Stream Encrypting Key, which in turn provides access
+to the protected payload of SRT data packets. The KEK has to be at least as long as the SEK.
 
-The KEK is generated by a password-based key generation function (PBKDF2)<!-- [PCKS5] -->,
-which uses the passphrase, a number of iterations (2048), a keyed-hash (HMAC-SHA1), and
-a length KEYLEN. The PBKDF2 function hashes the passphrase to make a long string,
+The KEK is generated by a password-based key generation function v2.0 (PBKDF2) {{RFC2898}},
```

Comment:
My understanding is that PBKDF2 is the second of two functions defined in RFC 2898, not "version 2.0" (i.e. in future, it might happen that there is a version x.y of PBKDF2). Perhaps just drop the "v2.0" (the PBKDF2 in parentheses is specific enough).

Reply:
Yes, it looks like you are right. Version 2.0 is the version of the `PKCS #5`, which defines the PBKFD2 function.

Reply:
Removed all v2.0 mentions.